_posts/2019-04-10-imperatywnie-vs-funkcyjnie.md

```diff
@@ -0,0 +1,136 @@
+---
+layout:    post
+title:     "Wstęp do programowania funkcyjnego"
+date:      2019-04-10 9:00:00 +0100
+published: false
+author:    Łukasz Sroczyński
+tags:
+    - backend
+    - kotlin
+    
+---
+
+Programowanie funkcyjne w ostatnich latach stało się bardzo popularne. Umożliwia nam to pisanie kodu, który jest czystszy, bezpieczniejszy, a przede wszystkim łatwo testowalny. Oczywiście FP ma również wady z czego jedną z nich może być zbyt duże pójście w abstrakcję co powoduje, że kod staje się niezrozumiały. Omówimy sobie w takim razie jakie są różnice pomiędzy imperatywnym podejściem, a tym funkcyjnym. 
+
+## Różnica między kodem imperatywnym, a funkcyjnym?
+Imperatywny kod to ten z którym najczęściej spotykamy się na początku naszej przygody z programowaniem. Jest to bowiem najbardziej naturalny sposób w jaki można pisać aplikację. Tworzymy tutaj ciąg instrukcji jaki nasz program wykonuje (step-by-step). Opisujemy dokładne czynności jakie muszą być wykonane podczas działania algorytmu. Podczas tych kroków zmieniamy stan systemu modyfikując go, a wynikiem końcowym jest zwrócona wartość lub też inny efekt.
+
+Cechy imperatywnego kodu: 
+
+* zmieniający się stan podczas każdej iteracji
+* kolejność wykonywania jest ważna (step-by-step)
+* wywołujemy funkcje, pętle, warunki
+
+Mówiąc o imperatywnym podejściu nie można nie wspomnieć o najbardziej popularnym z paragymatów, czyli obiektówce. Skupiamy się tutaj na hierarchii klas, enkapsulacji oraz wielu, wielu innych elementach przekraczających zakres tego wpisu. Generalnie myślimy w kontekście obiektów. Wbrew pozorom większość popularnych języków obiektowych wspiera/wywodzi się właśnie z imparatywnego/proceduralnego podejścia. Aby przejść z tego paradygmatu na ten bardziej matematyczny/funkcyjny trzeba zmienić myślenie. 
```

Comment:
"obiktowym" a nie "obiektówce" - używajmy nieco mniej luźnego języka

Reply:
"Skupiamy się tutaj na hierarchii klas, enkapsulacji" - bym się nie zgodził...

Reply:
"imparatywnego/proceduralnego" - mamy sobie wybrać? No i gdzie jest "programowanie strukturalne"? 
Uprośćmy to do "imperatywnego"

Reply:
"Aby przejść z tego paradygmatu na ten bardziej matematyczny/funkcyjny trzeba zmienić myślenie." - to zdanie trochę mi nie pasuje do tego paragrafu

Reply:
Usunąłem kawałek